Protected Constructor of class ...

With gcc it seams to be fine, but with clang (tried clang 6.0 and clang 8.0) it blames:

It seams to be a duplicate of #13 

```c++
In file included from /home/fins/Development/json-repository/plugins/ApplicationAdapter/src/ApplicationAdapter.cpp:3:
[build] In file included from /usr/local/include/dbus-cxx-0.11/dbus-cxx.h:28:
[build] In file included from /usr/local/include/dbus-cxx-0.11/dbus-cxx/callmessage.h:19:
[build] In file included from /usr/local/include/dbus-cxx-0.11/dbus-cxx/message.h:26:
[build] /usr/local/include/dbus-cxx-0.11/dbus-cxx/messageiterator.h:550:17: error: calling a protected constructor of class 'DBus::ErrorInvalidTypecast'
[build]           throw ErrorInvalidTypecast( s.c_str() );
[build]                 ^
[build] /usr/local/include/dbus-cxx-0.11/dbus-cxx/error.h:347:18: note: declared protected here
[build]   DBUSCXX_ERROR( ErrorInvalidTypecast, "Invalid Typecast" );
[build]                  ^
```

Why not making ctor public?

--- a/dbus-cxx/error.h
+++ b/dbus-cxx/error.h
@@ -98,10 +98,9 @@ namespace DBus
 
 #define DBUSCXX_ERROR( CPPTYPE, DBUS_ERROR_CODE )            \
   class CPPTYPE : public Error {                             \
-    protected:                                               \
+    public:                                                  \
     CPPTYPE( const char* message = NULL )                    \
         : Error( DBUS_ERROR_CODE, message ) {}               \
-    public:                                                  \
       typedef DBusCxxPointer<CPPTYPE> pointer;               \
       static pointer create( const char* message = NULL ) {  \
         return pointer( new CPPTYPE(message) );              \

